crates/node/src/vmm/qemu.rs

```diff
@@ -189,7 +189,10 @@ impl Provider for Qemu {
 
         // Update the child process field.
         let qemu_vm_handle = &mut self.vm_registry.get_mut(&cid).unwrap();
-        let mut cmd = Command::new("/usr/bin/qemu-system-x86_64");
+
+        // Resolve the QEMU system command
+        let qemu_path = which::which("qemu-system-x86_64")?;
+        let mut cmd = Command::new(qemu_path);
```

Comment:
Instead of resolving the path on every VM startup, could this be done when constructing the QEMU provider at `new()` ?